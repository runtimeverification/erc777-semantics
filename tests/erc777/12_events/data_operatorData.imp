test.preamble();

// Functions to test: send, operatorSend, burn, operatorBurn, operatorMint.
//Hooks: all allowed, all accepting. Send hooks for senders, receive hooks for receivers.

//Other: one single operation for each
//Parameters: non-empty data, operatorData for each operation.

test.registerTokensToSendHook(5, 1);
test.registerTokensReceivedHook(6, 1);

test.makeCaller(5);
test.send(6, 1, "d_send");
test.burn(10);

test.makeCaller(8);
test.operatorSend(5, 6, 10, "data_opSend", "operatorData_opSend");
test.operatorBurn(5, 10, "operatorData_burn");
test.operatorMint(6, 100, "data_opMint", "operatorData_opMint");

printLogs;
