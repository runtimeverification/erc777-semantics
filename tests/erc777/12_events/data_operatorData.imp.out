Registering holder 5 with hook 1
Registering receiver 6 with hook 1
Changing the caller to 5
5 is sending 1 to 6 (data:d_send)
tokensToSend: Hook address = 1, Operator = 5, From = 5, To = 6, Value = 1, Data = d_send, OperatorData = 
tokensToSend: sender balance: 50
tokensToSend: receiver balance: 60
tokensReceived: Hook address = 1, Operator = 5, From = 5, To = 6, Value = 1, Data = d_send, OperatorData = 
tokensReceived: sender balance: 49
tokensReceived: receiver balance: 61
Balance of 5 is 49
Balance of 6 is 61
Burning 10 units from 5
tokensToSend: Hook address = 1, Operator = 5, From = 5, To = 0, Value = 10, Data = , OperatorData = 
tokensToSend: sender balance: 49
tokensToSend: receiver balance: 0
Balance of 5 is 39
Total supply is 540
Changing the caller to 8
8 is sending 10 from 5 to 6 (data:"data_opSend"),  (operatorData:"operatorData_opSend")
tokensToSend: Hook address = 1, Operator = 8, From = 5, To = 6, Value = 10, Data = data_opSend, OperatorData = operatorData_opSend
tokensToSend: sender balance: 39
tokensToSend: receiver balance: 61
tokensReceived: Hook address = 1, Operator = 8, From = 5, To = 6, Value = 10, Data = data_opSend, OperatorData = operatorData_opSend
tokensReceived: sender balance: 29
tokensReceived: receiver balance: 71
Balance of 5 is 29
Balance of 6 is 71
8 is burning 10 units from 5 (operatorData:"operatorData_burn")
tokensToSend: Hook address = 1, Operator = 8, From = 5, To = 0, Value = 10, Data = , OperatorData = operatorData_burn
tokensToSend: sender balance: 29
tokensToSend: receiver balance: 0
Balance of 5 is 19
Total supply is 530
8 is minting 100 for 6 (data:"data_opMint"),  (operatorData:"operatorData_opMint")
tokensReceived: Hook address = 1, Operator = 8, From = 0, To = 6, Value = 100, Data = data_opMint, OperatorData = operatorData_opMint
tokensReceived: sender balance: 0
tokensReceived: receiver balance: 171
Balance of 6 is 171
Total supply is 630

Events:
Minted(8, 10, 100, , init)
Transfer(0, 10, 100)
Minted(8, 9, 90, , init)
Transfer(0, 9, 90)
Minted(8, 8, 80, , init)
Transfer(0, 8, 80)
Minted(8, 7, 70, , init)
Transfer(0, 7, 70)
Minted(8, 6, 60, , init)
Transfer(0, 6, 60)
Minted(8, 5, 50, , init)
Transfer(0, 5, 50)
Minted(8, 4, 40, , init)
Transfer(0, 4, 40)
Minted(8, 3, 30, , init)
Transfer(0, 3, 30)
Minted(8, 2, 20, , init)
Transfer(0, 2, 20)
Minted(8, 1, 10, , init)
Transfer(0, 1, 10)
Sent(5, 5, 6, 1, d_send, )
Transfer(5, 6, 1)
Burned(5, 5, 10, )
Transfer(5, 0, 10)
Sent(8, 5, 6, 10, data_opSend, operatorData_opSend)
Transfer(5, 6, 10)
Burned(8, 5, 10, operatorData_burn)
Transfer(5, 0, 10)
Minted(8, 6, 100, data_opMint, operatorData_opMint)
Transfer(0, 6, 100)
